backend/src/media/backends/azure-backend.ts

```diff
@@ -1,83 +1,107 @@
 /*
- * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ * SPDX-FileCopyrightText: 2024 The HedgeDoc developers (see AUTHORS file)
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 import {
+  BlobSASPermissions,
   BlobServiceClient,
   BlockBlobClient,
   ContainerClient,
+  generateBlobSASQueryParameters,
+  StorageSharedKeyCredential,
 } from '@azure/storage-blob';
 import { Inject, Injectable } from '@nestjs/common';
+import { FileTypeResult } from 'file-type';
 
 import mediaConfiguration, { MediaConfig } from '../../config/media.config';
 import { MediaBackendError } from '../../errors/errors';
 import { ConsoleLoggerService } from '../../logger/console-logger.service';
 import { MediaBackend } from '../media-backend.interface';
-import { BackendData } from '../media-upload.entity';
 import { BackendType } from './backend-type.enum';
 
 @Injectable()
 export class AzureBackend implements MediaBackend {
   private config: MediaConfig['backend']['azure'];
   private client: ContainerClient;
+  private readonly credential: StorageSharedKeyCredential;
 
   constructor(
     private readonly logger: ConsoleLoggerService,
     @Inject(mediaConfiguration.KEY)
     private mediaConfig: MediaConfig,
   ) {
     this.logger.setContext(AzureBackend.name);
-    this.config = mediaConfig.backend.azure;
-    if (mediaConfig.backend.use === BackendType.AZURE) {
+    this.config = this.mediaConfig.backend.azure;
+    if (this.mediaConfig.backend.use === BackendType.AZURE) {
       // only create the client if the backend is configured to azure
       const blobServiceClient = BlobServiceClient.fromConnectionString(
         this.config.connectionString,
       );
+      this.credential =
+        blobServiceClient.credential as StorageSharedKeyCredential;
       this.client = blobServiceClient.getContainerClient(this.config.container);
     }
   }
 
   async saveFile(
+    uuid: string,
     buffer: Buffer,
-    fileName: string,
-  ): Promise<[string, BackendData]> {
+    fileType: FileTypeResult,
+  ): Promise<null> {
     const blockBlobClient: BlockBlobClient =
-      this.client.getBlockBlobClient(fileName);
+      this.client.getBlockBlobClient(uuid);
     try {
-      await blockBlobClient.upload(buffer, buffer.length);
-      const url = this.getUrl(fileName);
-      this.logger.log(`Uploaded ${url}`, 'saveFile');
-      return [url, null];
+      await blockBlobClient.upload(buffer, buffer.length, {
+        blobHTTPHeaders: {
+          blobContentType: fileType.mime,
+        },
+      });
+      this.logger.log(`Uploaded file ${uuid} to Azure`, 'saveFile');
```

Comment:
```suggestion
      this.logger.log(`Uploaded file ${uuid}`, 'saveFile');
```

We don't do this in other backends as well…